AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -258,12 +362,18 @@ - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
   
   stack.children = children;
   
-  return stack;
+  if (self.backgroundImageNode.image) {
+      return [ASBackgroundLayoutSpec backgroundLayoutSpecWithChild:stack
+                                                        background:self.backgroundImageNode];
+  } else {
+      return stack;
+  }
 }
 
 - (void)layout
 {
   [super layout];
+  self.backgroundImageNode.hidden = self.backgroundImageNode.image == nil;
```

Comment:
In an ideal world, we'd not allocate and add all of the subnodes until we know they're needed - but there are some complexities with doing that if e.g. the title were set in the background after initial loading of the views / layers.  This is probably fine for now especially with layer-backing now on (less in-app CPU time to create 3 layers than 2 views - by a mile), but the layers really add up in terms of CATransaction cost, and the much-harder-to-measure render server per frame recursion cost to set up the OpenGL compositing of the frame.  Even if hidden, they add measurable cost to the hierarchy scanning.